src/Uno.UWP/UI/Composition/CompositionObject.cs

```diff
@@ -25,17 +27,184 @@ internal CompositionObject(Compositor compositor)
 
 		public CoreDispatcher Dispatcher => CoreDispatcher.Main;
 
+		public string? Comment { get; set; }
+
 		public void StartAnimation(string propertyName, CompositionAnimation animation)
 		{
 			StartAnimationCore(propertyName, animation);
 		}
 
+		public void StopAnimation(string propertyName)
+		{
+
+		}
+
 		internal virtual void StartAnimationCore(string propertyName, CompositionAnimation animation) { }
 
-		public void StopAnimation(string propertyName)
+		internal void AddContext(CompositionObject context, string? propertyName)
 		{
+			_contextStore.AddContext(context, propertyName);
 		}
 
-		public string? Comment { get; set; }
+		internal void RemoveContext(CompositionObject context, string? propertyName)
+		{
+			_contextStore.RemoveContext(context, propertyName);
+		}
+
+		private protected void SetProperty<T>(ref T field, T value, [CallerMemberName] string? propertyName = null)
+		{
+			var fieldCO = field as CompositionObject;
+			var valueCO = value as CompositionObject;
+			if (fieldCO != null || value != null)
+			{
+				OnCompositionPropertyChanged(fieldCO, valueCO, propertyName);
+			}
+
+			field = value;
+
+			OnPropertyChanged(propertyName, false);
+		}
+
+		private protected void OnChanged() => OnPropertyChanged(null, false);
+
+		private protected void OnCompositionPropertyChanged(CompositionObject? oldValue, CompositionObject? newValue) => OnCompositionPropertyChanged(oldValue, newValue, null);
+
+		private protected void OnCompositionPropertyChanged(CompositionObject? oldValue, CompositionObject? newValue, string? propertyName)
+		{
+			if (oldValue != null)
+			{
+				oldValue.RemoveContext(this, propertyName);
+			}
+
+			if (newValue != null)
+			{
+				newValue.AddContext(this, propertyName);
+			}
+		}
+
+		private protected void OnPropertyChanged(string? propertyName, bool isSubPropertyChange)
+		{
+			OnPropertyChangedCore(propertyName, isSubPropertyChange);
+			_contextStore.RaiseChanged();
+		}
+
+		private protected virtual void OnPropertyChangedCore(string? propertyName, bool isSubPropertyChange)
+		{
+
+		}
+
+		private class ContextStore
```

Comment:
Maybe this could be moved to a separate file with `partial` (like `CompositionObject.Store.cs`). Just to have it separated a bit.

Reply:
Done - moved to CompositionObject.Context.cs.